src/System.Management.Automation/engine/Utils.cs

```diff
@@ -478,13 +478,11 @@ internal static string GetWindowsPowerShellVersionFromRegistry()
         }
 #endif
 
-        internal static string DefaultPowerShellAppBase => GetApplicationBase(DefaultPowerShellShellID);
+        internal static string DefaultPowerShellAppBase => GetApplicationBase();
 
-        internal static string GetApplicationBase(string shellId)
+        internal static string GetApplicationBase()
         {
-            // Use the location of SMA.dll as the application base.
-            Assembly assembly = typeof(PSObject).Assembly;
-            return Path.GetDirectoryName(assembly.Location);
+            return AppContext.BaseDirectory ?? Path.GetDirectoryName(Environment.ProcessPath);
```

Comment:
Some remoting tests were failing.
So, it seems `Environment.ProcessPath` doesn't work correctly in WSMan remoting scenario. I'm afraid we need to change this to `AppContext.BaseDirectory ?? Path.GetDirectoryName(typeof(PSObject).Assembly.Location);`

I wonder what `[System.Reflection.Assembly]::GetExecutingAssembly()` will return in case of single-exe.
@SteveL-MSFT can you please point me to your single-exe branch? Thanks

Reply:
Environment.ProcessPath is for NativeAOT scenario as implementation in .Net repo says.

WSMan remoting - it is "native host with managed code" scenario if I understand correctly?

Reply:
Maybe useful about single exe https://learn.microsoft.com/en-us/dotnet/core/deploying/single-file/overview?tabs=cli#api-incompatibility

> Assembly.Location Returns an empty string.

Reply:
@daxian-dbw here's my experimental branch for single-exe https://github.com/stevel-msft/powershell/tree/single-trim

Reply:
Had to add back assembly.location for remoting to work correctly